internal/controllers/s3userclaim/provisioner.go

```diff
@@ -131,6 +133,81 @@ func (r *Reconciler) ensureReadonlySubuser(ctx context.Context) (*ctrl.Result, e
 	return subreconciler.ContinueReconciling()
 }
 
+func (r *Reconciler) ensureOtherSubusers(ctx context.Context) (*ctrl.Result, error) {
```

Comment:
This `ensure` is a little bit misleading as we also remove some sub-users here. Consider a name like `syncSubusersList` or event splitting it into 2/3 functions. One tagging the subusers, one removing the unwanted ones, and the third one creating/updating the desired ones.

Also, I think a comment describing what this function does is indeed helpful. For example:

````
<function name> Iterates over the existing subusers. It removes any of them not included in the current spec and ensures all of the subusers in the current spec are created in Ceph.
````

Reply:
Done